components/org.wso2.carbon.identity.sso.saml.ui/src/main/java/org/wso2/carbon/identity/sso/saml/ui/SAMLSSOUIUtil.java

```diff
@@ -81,4 +87,51 @@ public static String getSafeInput(HttpServletRequest request, String parameter)
         return filteredProviders;
     }
 
+    public static boolean isResponseSigningEnabled(boolean isSpEdit, SAMLSSOServiceProviderDTO provider) {
+
+        if (isSpEdit) {
+            if (provider != null) {
```

Comment:
can the provider be null? since this is an update.

In case of provider==null, are we throwing an error in somewhere else?

Reply:
I'm not certains whether we could assume provider is always not null hence the null check is added as a safeguard.